### ISSUE TITLE
Interaction between ChannelError and Select

### ISSUE THREAD

**Issue by ytoh** [softwaremill/jox#73]
Hello!

I have built an abstraction that internally use `Channels`. I am now designing the exception handling for it. When I read from a single channel everything is clear, I can translate `ChannelError` to whatever I need and provide context of my abstraction.

Where I have a problem is when I use `Select#select`. In this case, when an error occurs I do not know from which channel it originates and I have to go through all of them to check whether I have the correct one. 

Would it be possible to either expose the channel on `ChannelClosed` or allow some sort of metadata to be attached to a channel that would then be accessible on `ChannelClosed`? Or is there a better way of doing such a thing?

I am willing to work on it myself if you deem such a feature desirable.

Thanks

**Comment by adamw** [comment]
Yes, I think it's the same as https://github.com/softwaremill/jox/issues/64?
If you could attempt a PR, I'll be happy to review.
